tests/selftest.js

```diff
@@ -27,10 +27,24 @@ service = server.listen(testServerPort, function(request, response) {
     } else {
         var headers = {};
         var binMode = false;
-        if (/js$/.test(pageFile)) {
+        if (/html$/.test(pageFile)) {
+            headers['Content-Type'] = "text/html";
+        }
+        else if (/js$/.test(pageFile)) {
             headers['Content-Type'] = "application/javascript";
         }
+        else if (/json$/.test(pageFile)) {
+            headers['Content-Type'] = "application/json";
+        }
+        else if (/txt$/.test(pageFile)) {
+            headers['Content-Type'] = "text/plain";
+        }
         else if (/png$/.test(pageFile)) {
+            headers['Content-Type'] = "image/png";
+            binMode = true;
+        }
```

Comment:
All this can probably be refactored using a map, eg. `{"json": "application/json", etc.}`